File: app/src/main/java/com/pr0gramm/app/ui/ConservativeLinearLayoutManager.java
package com.pr0gramm.app.ui;

import android.content.Context;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.view.View;

public class ConservativeLinearLayoutManager extends LinearLayoutManager {
    public ConservativeLinearLayoutManager(Context context, int orientation, boolean reverseLayout) {
        super(context, orientation, reverseLayout);
    }

    @Override
    public void onMeasure(RecyclerView.Recycler recycler, RecyclerView.State state,
                          int widthSpec, int heightSpec) {

        int newHeightSpec = View.MeasureSpec.makeMeasureSpec(
                View.MeasureSpec.getSize(heightSpec),
                View.MeasureSpec.UNSPECIFIED);

        super.onMeasure(recycler, state, widthSpec, newHeightSpec);
    }
}

File: app/src/main/java/com/pr0gramm/app/ui/ConservativeLinearLayoutManager.kt
package com.pr0gramm.app.ui

import android.content.Context
import android.view.View

class ConservativeLinearLayoutManager(context: Context, orientation: Int, reverseLayout: Boolean) :
        androidx.recyclerview.widget.LinearLayoutManager(context, orientation, reverseLayout) {

    override fun onMeasure(recycler: androidx.recyclerview.widget.RecyclerView.Recycler, state: androidx.recyclerview.widget.RecyclerView.State,
                           widthSpec: Int, heightSpec: Int) {

        val newHeightSpec = View.MeasureSpec.makeMeasureSpec(
                View.MeasureSpec.getSize(heightSpec),
                View.MeasureSpec.UNSPECIFIED)

        super.onMeasure(recycler, state, widthSpec, newHeightSpec)
    }
}